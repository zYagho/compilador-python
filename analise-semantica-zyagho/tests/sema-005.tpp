{Erro: Chamada à função 'func' com número de parâmetros menor que o declarado.}
{Erro: Chamada à função 'func' com número de parâmetros maior que o declarado.}
{Aviso: Chamada à função 'func' com Coerção Implícita do valor do argumento '10.5' tipo 'flutuante' diferente do parâmetro declarado 'inteiro'.}
{Aviso: Chamada à função 'func' com Coerção Implícita do valor do argumento '10.5' tipo 'flutuante' diferente do parâmetro declarado 'inteiro'.}
{Aviso: Chamada à função 'func' com Coerção Implícita do valor do argumento 'b' tipo 'flutuante' diferente do parâmetro declarado 'inteiro'.}
{Aviso: Variável 'a' declarada e inicializada, mas não utilizada}
{Erro: Função 'principal' deveria retornar 'inteiro', mas retorna 'vazio'}

inteiro func(inteiro: x, inteiro: y)
	retorna(x + y)
fim

inteiro principal()
	inteiro: a
	flutuante: b
	b := 10.5

	a := func(10)
	a := func(10,20,30)
	a := func(10.5,20)
	a := func(10.5,b)
fim
