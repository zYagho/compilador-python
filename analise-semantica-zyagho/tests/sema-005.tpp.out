ERR-SEM-CALL-FUNC-WITH-FEW-ARGS
ERR-SEM-CALL-FUNC-WITH-MANY-ARGS
ERR-SEM-FUNC-RET-TYPE-ERROR
WAR-SEM-ATR-DIFF-TYPES-IMP-COERC-OF-FUNC-ARG
WAR-SEM-ATR-DIFF-TYPES-IMP-COERC-OF-FUNC-ARG
WAR-SEM-ATR-DIFF-TYPES-IMP-COERC-OF-FUNC-ARG
WAR-SEM-VAR-DECL-INIT-NOT-USED
ERR-SEM-FUNC-RET-TYPE-ERROR
